corda/corda-api
# CORE-17430 Modify the utxo_transaction_sources table by removing unused columns and renaming

### Overview

The `created` and `is_ref_input` columns are unnecessary to have in the `utxo_transaction_sources` table.
The created column will be the same as in utxo_transaction_component table and it’s a 1:1 mapping.
Whether a state is an input or a reference can be found out by using the groupIdx column rather than having an extra is_ref_input column.

Also rename the ref_transaction_id and ref_leaf_idx columns as those are quite confusing.

### Test plan
- Run the combined worker with Postgres (5.0) then build the VNode CLI plugin (5.1), run the migration that generates a SQL file. Run the SQL file in the 5.0 DB. Check that the fields are present and data was migrated properly.

- Run a combined worker with Postgres (5.1) and make sure no errors during startup / e2e tests. Check the DB that transactions are actually stored and the new fields are populated.
